File: solutions/elixir/pe_001_02.exs
# Project Euler <https://projecteuler.net>
# Problem: 1
# Solution: 2
# Author: Sean Broderick <hakutsuru@mac.com>
# Date: 2015-09-25


# this is cheap, but... gonna use erlang

{threshold, _} = Integer.parse(List.first(System.argv))

sum3 = Enum.sum :lists.seq(3,threshold,3)

already_included? = fn(n) -> (rem(n, 3) == 0) end
multiples_of_5 = for n <- :lists.seq(5,threshold,5), !already_included?.(n), do: n
sum5 = Enum.sum multiples_of_5

sum = sum3 + sum5
IO.inspect sum


# Result (sum):
# /opt/euler/elixir$ elixir pe_001_02.exs 9
# 23
# /opt/euler/elixir$ elixir pe_001_02.exs 999
# 233168
# /opt/euler/elixir$ elixir pe_001_02.exs 999999
# 233333166668

File: solutions/erlang/pe_001_02.erl
% Project Euler <https://projecteuler.net>
% Problem: 1
% Solution: 2
% Author: Sean Broderick <hakutsuru@mac.com>
% Date: 2015-06-10


-module(pe_001_02).
-export([sum/1]).

sum(Threshold) ->
   Multiples3 = [X || X <- lists:seq(3,Threshold,3)],
   Sum3 = lists:sum(Multiples3),
   Multiples5 =  [X || X <- lists:seq(5,Threshold,5), (X rem 3 /= 0)],
   Sum5 = lists:sum(Multiples5),
   Sum3 + Sum5.


% Result (sum):
%/opt/euler/erlang$ erl
% 1> c(pe_001_02).
% {ok,pe_001_02}
% 2> pe_001_02:sum(9).
% 23
% 3> pe_001_02:sum(999).
% 233168
% 4> pe_001_02:sum(999999).
% 233333166668
